pkg/certificate/controller_test.go

```diff
@@ -175,161 +172,184 @@ var _ = Describe("Certificates controller", func() {
-						Expect(currentResult.RequeueAfter).To(Equal(mgr.elapsedToRotateCAFromLastDeadline()), "should schedule new Reconcile after service cert rotation to rotate service cert again")
+						Expect(currentResult.RequeueAfter).To(Equal(mgr.elapsedToRotateServicesFromLastDeadline()), "should schedule new Reconcile after service cert rotation to rotate service cert again")
+
+						certs, err := triple.ParseCertsPEM(currentTLS.serviceCertificate)
+						Expect(err).To(Succeed(), "should succeed parsing service certificates")
+						Expect(certs).To(HaveLen(1), "should have cleanup overlap service certs")
 					})
-					Context("and called at previous RequeueAfter (ca cert rotation deadline)", func() {
+
+					Context("and called at previous RequeueAfter (second service cert rotation deadline)", func() {
 						BeforeEach(func() {
-							previousTLS = currentTLS
-							previousResult = currentResult
-							// Emulate controller-runtime timer by adding previous RequeueAfter values to previousNow
-							now = mgr.now().Add(previousResult.RequeueAfter)
-							mgr.now = func() time.Time { return now }
-							triple.Now = mgr.now
-							var err error
-							By("Reconcile at ca cert rotation deadline")
-							currentResult, err = mgr.Reconcile(reconcile.Request{})
-							Expect(err).To(Succeed(), "should success reconciling")
-							currentTLS = getTLS()
+							backToTheFuture("Reconcile at second service cert rotation deadline", currentResult.RequeueAfter)
 						})
-						It("should rotate CA and service certs and return new deadline", func() {
+						It("should rotate service cert/key and return a new deadline", func() {
 							Expect(currentTLS.serviceCertificate).ToNot(Equal(previousTLS.serviceCertificate), "should have do TLS cert rotation")
 							Expect(currentTLS.servicePrivateKey).ToNot(Equal(previousTLS.servicePrivateKey), "should have do a TLS key rotation")
 							Expect(currentTLS.serviceSecretAnnotations).To(Equal(previousTLS.serviceSecretAnnotations), "should containe same secret annotations")
-							Expect(currentTLS.caBundle).ToNot(Equal(previousTLS.caBundle), "should have rotate CABundle ")
-							Expect(currentTLS.caCertificate).ToNot(Equal(previousTLS.caCertificate), "should have rotate CA certificate")
-							Expect(currentTLS.caPrivateKey).ToNot(Equal(previousTLS.caPrivateKey), "should have rotate CA key rotation")
-
-							elapsedForCleanup, err := mgr.earliestElapsedForCleanup()
-							Expect(err).To(Succeed(), "should succeed calculating earliestElapsedForCleanup")
-							Expect(currentResult.RequeueAfter).To(Equal(elapsedForCleanup), "Reconcile at rotate should schedule next Reconcile to do the CA overlapping cleanup")
-
-							cas, err := triple.ParseCertsPEM(currentTLS.caBundle)
-							Expect(err).To(Succeed(), "should succeed parssing caBundle")
-							Expect(cas).To(HaveLen(2), "should overlap CAs")
+							Expect(currentTLS.caBundle).To(Equal(previousTLS.caBundle), "shouldn't have rotate CABundle ")
```

Comment:
These multiple expects are repeating themselves, what do you think about moving them to 4 functions:
- `CheckServiceRotated(previousTLS, currentTLS)`
- `CheckServiceNotRotated(previousTLS, currentTLS)`
- `CheckCARotated(previousTLS, currentTLS)`
- `CheckCANotRotated(previousTLS, currentTLS)`